src/rust/scheduler/scheduler.rs

```diff
@@ -356,8 +344,39 @@ mod tests {
         Ok(())
     }
 
+    /// Tests if consecutive tasks are not assigned the same task id.
+    #[test]
+    fn insert_consecutive_creates_unique_task_ids() -> Result<()> {
+        let scheduler: Scheduler = Scheduler::default();
+
+        // Create and run a task.
+        let task: DummyTask = DummyTask::new(String::from("testing"), Box::pin(DummyCoroutine::new(0)));
+        let handle: TaskHandle = match scheduler.insert(task) {
+            Some(handle) => handle,
+            None => anyhow::bail!("insert() failed"),
+        };
+        let task_id: u64 = handle.clone().get_task_id();
+        scheduler.poll();
+
+        // Ensure that the first task has completed.
+        crate::ensure_eq!(handle.has_completed(), true);
+
+        // Create another task.
+        let task2: DummyTask = DummyTask::new(String::from("testing"), Box::pin(DummyCoroutine::new(0)));
+        let handle2: TaskHandle = match scheduler.insert(task2) {
+            Some(handle) => handle,
+            None => anyhow::bail!("insert() failed"),
+        };
+        let task_id2: u64 = handle2.get_task_id();
+
+        // Ensure that the second task has a unique id.
+        crate::ensure_neq!(task_id2, task_id);
+
+        Ok(())
+    }
+
     #[test]
-    fn scheduler_poll_once() -> Result<()> {
+    fn poll_once_with_one_small_task_completes_it() -> Result<()> {
```

Comment:
Could you add a Rust Doc comment for this test, like we have for other tests in this module?